File: codigo/planejador/src/main/java/inteli/cc6/Middleware/Middleware.java
package inteli.cc6.Middleware;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
/**
 * The Middleware class implements the WebMvcConfigurer interface to provide configuration for CORS (Cross-Origin Resource Sharing) mappings.
 */
public class Middleware implements WebMvcConfigurer {

    /**
     * Configures the CORS mappings.
     *
     * @param registry the CorsRegistry object used to register CORS mappings
     */
    @Override
    public void addCorsMappings(CorsRegistry registry) {
        registry.addMapping("/")
                .allowedOrigins("http://127.0.0.1:3000")
                .allowedMethods("*")
                .allowedHeaders("*");
    }
}

File: codigo/back-end/src/middleware/middleware.js
/**
 * @class Middleware
 * @description Represents a middleware class used for handling CORS (Cross-Origin Resource Sharing).
 */
class Middleware {
    /**
     * @static
     * @function cors
     * @description Returns the CORS configuration object.
     * @returns {object} - The CORS configuration object.
     */
    static cors() {
      return {
        origin: ''
      };
    }
  }
  
  export default Middleware;
  